src/api/services/schemaService.ts

```diff
@@ -36,3 +36,17 @@ export async function deleteSchema(id: string) {
     return data;
   }
 }
+export async function updateSchema(id: string, data: ISchema) {
+  if (typeof data.slug !== "string" || !Array.isArray(data.fields))
+    throw new ValidationError("Slug must be a string and fields must be an array");
+  if (data.slug === "" || data.fields.length === 0)
+    throw new ValidationError("Schema must have a slug and at least one field");
```

Comment:
Request body validation should be happening in controller functions, not the service layer